executor/executor.go

```diff
@@ -767,23 +767,41 @@ func (e *MaxOneRowExec) Next(goCtx goctx.Context) (Row, error) {
 type UnionExec struct {
 	baseExecutor
 
-	finished atomic.Value
-	resultCh chan *execResult
-	rows     []Row
-	cursor   int
-	wg       sync.WaitGroup
-	closedCh chan struct{}
+	stopFetchData atomic.Value
+	resultCh      chan *execResult
+	rows          []Row
+	cursor        int
+	wg            sync.WaitGroup
+
+	// For chunk execution.
+	resoucePool chan *executorResult
+	resultPool  chan *executorResult
+	initialized bool
 }
 
 type execResult struct {
 	rows []Row
 	err  error
 }
 
-func (e *UnionExec) waitAllFinished() {
+type executorResult struct {
+	chk *chunk.Chunk
+	err error
```

Comment:
How about adding a channel in executorResult to reuse the chunk and remove `resourcePool`?

Reply:
I'll think about it